### ISSUE TITLE
Add a `--simulate-with-verilator` flag to `main.rkt`

### ISSUE THREAD

**Issue by bkushigian** [gussmith23/lakeroad#85]
It would be cool to be able to double check that a generated instruction is correct with verilator. This would act as a sort of dynamic end to end test

Gus's notes:

We're at the point where this would both be pretty easy (because we have a script for validation with Verilator that can just be called) and very useful (it would clean up both integration tests and the evaluation).

Tasks:
- [ ] Call the [simulate with Verilator script](https://github.com/uwsampl/lakeroad/blob/main/bin/simulate_with_verilator.py) from [main.rkt](https://github.com/uwsampl/lakeroad/blob/main/bin/main.rkt)
  We'll need to pass through flags for the Verilator script. As a rough first pass, we can just have an argument `--simulate-with-verilator-flags` which takes a string of all the flags to pass to the script.
- [ ] Convert [integration tests](https://github.com/uwsampl/lakeroad/tree/main/integration_tests/lakeroad) over to using the new flag, rather than explicitly calling the script ([example here](https://github.com/uwsampl/lakeroad/blob/2c6109a06c8f2cb2cc059ec92e5cf728c99b498a/integration_tests/lakeroad/xilinx_mulsub_1_stage_unsigned_14_bit.sv#L21-L46))
- [ ] Convert evaluation over to using this flag
  This doesn't need to be done before merging, but we should open an issue for it on the eval. The gist is: instead of [creating a task to run Lakeroad](https://github.com/uwsampl/lakeroad-evaluation/blob/7fb3a532085c0a6d47b3ed313bedf5ea86f568f7/python/robustness_experiments.py#L1120-L1150) and then [a separate task for running Verilator,](https://github.com/uwsampl/lakeroad-evaluation/blob/7fb3a532085c0a6d47b3ed313bedf5ea86f568f7/python/robustness_experiments.py#L1154-L1191) we can just pass the new flag to Lakeroad.

**Comment by gussmith23** [comment]
We're at the point where this would both be pretty easy (because we have a script for validation with Verilator that can just be called) and very useful (it would clean up both integration tests and the evaluation).

**Comment by gussmith23** [comment]
## 2024-11-22

Finally going to close this issue

Current issue here: it seems like Racket doesn't want to parse flags where the value of the flag also looks like a flag. This seems wrong---it really feels like there should be a way around this. Basically, it seems like having something like `--myflag="--myotherflag"` trips up the parser. I'm still not exactly sure how it's failing, because other experiments don't fail in expected ways. Like, passing a random flag into the script doesn't trigger the same error. It seems like the script ingests any argument that it doesn't recognize, and I guess it ignores it?

Actually specifically, it ignores a single unknown flag


Ah ok I figured it out. The base `--simulate-with-verilator` flag was expecting an argument when it shouldn't have.
